src/System Application/App/Table Information/src/TableInformationCacheImpl.Codeunit.al

```diff
@@ -144,8 +145,11 @@ codeunit 8700 "Table Information Cache Impl."
     end;
 
     procedure GetTableUrl(Company: Text; TableNo: Integer): Text
+    var
+        TableMetadata: Record "Table Metadata";
     begin
-        if Company = '' then
+        TableMetadata.Get(TableNo);
+        if not TableMetadata.DataPerCompany then
```

Comment:
What if Company is blank and the data is per company? Why cannot we always replace the blank company name with the current company?

This will also slow down the code since we will load the table metadata, which can be slow on the first call. I'm not sure how slow it is after the first call.

Reply:
This won't help.

The only thing that would help is to try a Company.Get(Company) to know if the `Company` text that was passed to the procedure is a valid company name.

I could then remove the check for the DataPerCompany. 

```al
if not CompanyRecord.Get(Company) then 
    Company := CompanyName(); 
```

Reply:
@pri-kise If the company exist or not it is a caller's responsibility to determine. This code should not fix it. If you ask for an invalid company URL, you should get what you ask. 

If we have a problem with blank company name, why cannot we replace it with current company? We know that it exists.

Reply:
The problem is not the Blank Company Name.

The problem is Company with the value of the label.
```al
CrossCompanyDataLbl: Label '(Cross-Company Data)'
```

I can change the code to check for the label `CrossCompanyDataLbl`.
....

Reply:
I would say if the table is a cross-company table, we should replace this string with current company name. We know that the current company exists. String comparison should be rather fast.

Reply:
@grobyns If the company name is blank, you will get the company defined in your user personalization. This could be OK if you want per-database data. So either omitting the company name or using the current company name should work IMO. Blank is probably better.

Reply:
I updated my code.
Now I change the Company Text back to an empty string if it equals to the CrossCompanyLbl